Source/WebCore/platform/graphics/cg/ImageUtilitiesCG.cpp

```diff
@@ -206,82 +206,97 @@ static void tryCreateNativeImageFromData(std::span<const uint8_t> data, std::opt
     });
 }
 
-static RefPtr<SharedBuffer> expandNativeImageToData(NativeImage& image, ASCIILiteral uti, std::span<const unsigned> lengths)
+static Vector<Ref<ShareableBitmap>> createBitmapsFromNativeImage(NativeImage& image, std::span<const unsigned> lengths)
 {
-    RetainPtr colorSpace = adoptCF(CGColorSpaceCreateWithName(kCGColorSpaceSRGB));
-    RetainPtr destinationData = adoptCF(CFDataCreateMutable(0, 0));
-    RetainPtr cfUTI = uti.createCFString();
-    RetainPtr destination = adoptCF(CGImageDestinationCreateWithData(destinationData.get(), cfUTI.get(), lengths.size(), nullptr));
+    Vector<Ref<ShareableBitmap>> bitmaps;
+    auto sourceColorSpace = image.colorSpace();
+    // The conversion could lead to loss of HDR contents.
+    auto destinationColorSpace = sourceColorSpace.supportsOutput() ? sourceColorSpace : DestinationColorSpace::SRGB();
     for (auto length : lengths) {
-        RetainPtr context = adoptCF(CGBitmapContextCreate(nullptr, length, length, 8, length * 4, colorSpace.get(), kCGImageAlphaPremultipliedLast));
-        if (!context)
-            return nullptr;
+        RefPtr bitmap = ShareableBitmap::createFromImageDraw(image, destinationColorSpace, { (int)length, (int)length }, ShareableBitmap::ShouldShrinkToFit::Yes);
```

Comment:
This can be written like this (See above):
``` 
RefPtr bitmap = ShareableBitmap::createFromImageDraw(image, destinationColorSpace, { (int)length, (int)length }, image.size());
```